src/box/txn.c

```diff
@@ -681,36 +686,39 @@ txn_commit_stmt(struct txn *txn, struct request *request)
 			++txn->n_applier_rows;
 		}
 	}
-	/*
-	 * If there are triggers, and they are not disabled, and
-	 * the statement found any rows, run triggers.
-	 * XXX:
-	 * - vinyl doesn't set old/new tuple, so triggers don't
-	 *   work for it
-	 * - perhaps we should run triggers even for deletes which
-	 *   doesn't find any rows
-	 */
-	if (stmt->space != NULL && stmt->space->run_triggers &&
-	    (stmt->old_tuple || stmt->new_tuple)) {
-		if (!rlist_empty(&stmt->space->before_replace)) {
-			/*
-			 * Triggers see old_tuple and that tuple
-			 * must remain the same
-			 */
-			stmt->does_require_old_tuple = true;
-		}
-		if (!rlist_empty(&stmt->space->on_replace)) {
-			/*
-			 * Triggers see old_tuple and that tuple
-			 * must remain the same
-			 */
-			stmt->does_require_old_tuple = true;
-
-			txn->space_on_replace_triggers_depth++;
-			int rc = trigger_run(&stmt->space->on_replace, txn);
-			txn->space_on_replace_triggers_depth--;
-			if (rc != 0)
-				goto fail;
+	struct space *space = stmt->space;
+	if (space != NULL) {
+		/*
+		 * If there are triggers, and they are not disabled, and
+		 * the statement found any rows, run triggers.
+		 * XXX:
+		 * - vinyl doesn't set old/new tuple, so triggers don't
+		 *   work for it
+		 * - perhaps we should run triggers even for deletes which
+		 *   doesn't find any rows
+		 */
+		if (space->run_triggers &&
+		    (stmt->old_tuple || stmt->new_tuple)) {
+			if (!rlist_empty(&space->before_replace)) {
+				/*
+				 * Triggers see old_tuple and that tuple
+				 * must remain the same
+				 */
+				stmt->does_require_old_tuple = true;
+			}
+			if (!rlist_empty(&space->on_replace)) {
+				/*
+				 * Triggers see old_tuple and that tuple
+				 * must remain the same
+				 */
+				stmt->does_require_old_tuple = true;
+				txn->space_on_replace_triggers_depth++;
+				int rc = trigger_run(&space->on_replace,
+						     txn);
+				txn->space_on_replace_triggers_depth--;
+				if (rc != 0)
+					goto fail;
+			}
```

Comment:
You don't seem to need this refactoring anymore, right?

Reply:
Yep. Sorry I missed thatn